### PR TITLE
Merge `tokio` in `platform_async`

### PR DESCRIPTION
**Pull request by TimeEngineer**
## Describe your changes

Instead to have `tokio` dependency everywhere, we can use `platform_async`
It also reduces the number of features
- before:
`bytes, default, fs, full, io-std, io-util, libc, macros, mio, net, num_cpus, parking_lot, process, rt, rt-multi-thread, signal, signal-hook-registry, socket2, sync, time, tokio-macros, windows-sys`
- after:
`bytes, default, fs, io-util, libc, mio, net, num_cpus, rt, rt-multi-thread, socket2, sync, time, windows-sys`

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
   
- [x] Ensure the commit history is sanitized
   
- [x] Give a meaningful title to your PR
- [x] Describe your changes
   
- [x] Link any related issue in the description
   
- [x] Link any dependent pull request in the description

If this PR adds a new feature or fixes a bug:

- [ ] I have added or updated relevant tests

If this PR modifies installation or updates packages:

- [ ] I have checked that installation works on all platforms (Windows, Linux, Mac ...)
    

- [ ] I have updated related documentations (installation commands, fix to specific problems ...)

If this PR is related to GUI:

- [ ] I have checked that GUIs affected by my changes are working properly.

If this PR affects the user:

- [ ] I have created a news fragment
   
- [ ] I have updated the user documentation

### CODE REVIEW

File: libparsec/crates/platform_storage/src/native/db/local_database.rs [hunk 1]
```diff
@@ -134,7 +135,7 @@ async fn new_sqlite_executor_from_path(
         }
     }?;
     if let Some(prefix) = path.parent() {
-        tokio::fs::create_dir_all(prefix).await.map_err(|e| {
+        tokio_create_dir_all(prefix).await.map_err(|e| {
```
Comment (touilleMan) [inline libparsec/crates/platform_storage/src/native/db/local_database.rs:138]:
```suggestion
        tokio_fs_create_dir_all(prefix).await.map_err(|e| {
```

File: libparsec/crates/platform_storage/src/native/db/local_database.rs [hunk 2]
```diff
@@ -193,7 +194,7 @@ impl LocalDatabase {
 
         /// Get the size of a file.
         async fn get_file_size(path: &Path) -> usize {
-            tokio::fs::metadata(path)
+            tokio_metadata(path)
```
Comment (touilleMan) [inline libparsec/crates/platform_storage/src/native/db/local_database.rs:197]:
```suggestion
            tokio_fs_metadata(path)
```

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: libparsec/crates/platform_async/src/lib.rs [hunk 3]
```diff
@@ -21,4 +21,13 @@ pub use platform::{
...
+pub use tokio::{
+    fs::{create_dir_all as tokio_create_dir_all, metadata as tokio_metadata},
+    runtime::Runtime,
+    sync::{
+        mpsc::unbounded_channel,
+        watch::{channel, Receiver},
+        Mutex, MutexGuard, RwLock, RwLockReadGuard, RwLockWriteGuard, TryLockError,
+    },
+    task::{spawn as tokio_spawn, spawn_blocking as tokio_spawn_blocking, JoinHandle},
+};
```
Comment (FirelightFlagboy) [inline libparsec/crates/platform_async/src/lib.rs:33]: Is the prefix `tokio` really useful ? I would prefer to not have it.

When we import something from `platform_async`, we already know that thing is async or related an async operation.

Comment (TimeEngineer) [reply libparsec/crates/platform_async/src/lib.rs:33]: `spawn` is already used and `create_dir_all` is confusing with the `std` one

Comment (FirelightFlagboy) [reply libparsec/crates/platform_async/src/lib.rs:33]: Than add the prefix to `JoinHandle` & `Runtime` to keep things homogeneous

**Comment by touilleMan** [comment]
replaced by #4942

**Pull request closed without merging**